backend/src/middleware/middleware.js

```diff
@@ -0,0 +1,48 @@
+const bcrypt = require('bcrypt');
+
+const { User } = require("../Schema")
+
+module.exports = function(req, res, next) {
+    const newUser = new User(req.body)
+
+    const errFill = new Error('Please fill all the input')
+    const errSave = new Error('Failled data saving')
+    if (newUser.username === '') {
+        next(errFill)
+    }
+    if (newUser.password === '') {
+        next(errFill)
+    }
+    if (newUser.email === '') {
+        next(errFill)
+    }
+    if (newUser.full_name === '') {
+        next(errFill)
+    }
+    if (newUser.role === '') {
+        next(errFill)
+    }
+    if (newUser.description === '') {
+        next(errFill)
+    }
+    if (newUser.gender === '') {
+        next(errFill)
+    }
+    if (newUser.birth_date === '') {
+        next(errFill)
+    }
+
+    const saltRounds = 10;
```

Comment:
This should be done inside the post handler. This middleware should only call User.validate() and call next() or return error.
Encrypting the password and calling user.save() should be done inside the POST handler.